### PR TITLE
idex decentralized exchange

### PR DESCRIPTION
**Pull request by jnorthrup**

### CODE REVIEW

**Comment by jnorthrup** [comment]
https://github.com/knowm/XChange/issues/2512#issuecomment-384779342 the deps look clean

**Comment by walec51** [comment]
as for some strange reason I cannot leave comments in the code then I'll have to do it here:

proxies should not be created in each method call: https://github.com/knowm/XChange/pull/2544/files#diff-0168def634f27171c78aef20ac7ed214R44

silent errors are pure evil - those exceptions should be just wrapped in a ExchangeException: https://github.com/knowm/XChange/pull/2544/files#diff-0168def634f27171c78aef20ac7ed214R70

you should try to follow this in new modules: https://github.com/knowm/XChange/wiki/New-Implementation-Best-Practices

**Comment by jnorthrup** [comment]
agreed on all points.

File: xchange-idex/src/main/java/org/knowm/xchange/idex/IdexSignature.java [hunk 1]
```diff
@@ -28,13 +29,13 @@
     ECKeyPair ecKeyPair;
     try (ByteArrayOutputStream sig_arr = new ByteArrayOutputStream()) {
 
-      if (IdexSignature.debugMe) {
+      if (debugMe) {
```
[the remark this answers is not in the document]
Comment (jnorthrup) [reply xchange-idex/src/main/java/org/knowm/xchange/idex/IdexSignature.java]: this data should not appear in any logs.  removed.

File: xchange-idex/src/main/java/org/knowm/xchange/idex/IdexTradeService.java [hunk 2]
```diff
@@ -78,48 +94,34 @@ public OpenOrders getOpenOrders() {
...
+    return getOpenOrders();
   }
 
   @Override
   public boolean cancelOrder(String s) {
-    CancelApi proxy =
-        RestProxyFactory.createProxy(
-            CancelApi.class, getIdexExchange().getExchangeSpecification().getSslUri());
-    return false;
+    throw new UnsupportedOperationException("Orders Api Unnavailable at this time");
```
[the remark this answers is not in the document]
Comment (jnorthrup) [reply xchange-idex/src/main/java/org/knowm/xchange/idex/IdexTradeService.java]: this change globally affects this set of files.

image

File: xchange-idex/pom.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    </organization>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>org.knowm.xchange</groupId>
+            <artifactId>xchange-core</artifactId>
+            <version>4.3.6-SNAPSHOT</version>
+        </dependency>
+        <dependency>
```
Comment (timmolter) [inline xchange-idex/pom.xml]: This should go in the root pom.xml under dependencyManagement. Here, just remove the version.

File: xchange-idex/src/main/java/org/knowm/xchange/idex/IdexExchange.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    }
+
+    public static CurrencyPair getCurrencyPair(String market) {
+      CurrencyPair currencyPair;
+      Iterator<String> syms = asList(market.split("_")).iterator();
+      currencyPair = new CurrencyPair(syms.next(), syms.next());
+      return currencyPair;
+    }
+
+    public static void main(String... args) throws IOException {
```
Comment (timmolter) [inline xchange-idex/src/main/java/org/knowm/xchange/idex/IdexExchange.java]: This doesn't belong here, perhaps move it to examples.

File: xchange-idex/pom.xml (hunk 3 above)
Comment (jnorthrup) [reply xchange-idex/pom.xml]: this was from huobi, the one before in the order, for consistency

File: xchange-idex/src/main/java/org/knowm/xchange/idex/IdexExchange.java (hunk 4 above)
Comment (jnorthrup) [reply xchange-idex/src/main/java/org/knowm/xchange/idex/IdexExchange.java]: will revisit this with a testnet example after some discussion.

File: xchange-idex/pom.xml (hunk 3 above)
Comment (timmolter) [reply xchange-idex/pom.xml]: I meant the org.web3j one :)

File: pom.xml [hunk 5]
```diff
@@ -255,6 +256,13 @@
                 <artifactId>logback-classic</artifactId>
                 <version>${version.qos.logback}</version>
             </dependency>
+
+            <dependency>
+                <groupId>org.web3j</groupId>
+                <artifactId>crypto</artifactId>
+                <version>3.2.0</version>
```
Comment (timmolter) [inline pom.xml]: Ant reason you chose version 3.2.0 when 3.4.0 is available?

**Pull request merged**